### PR TITLE
add table to record flagged posts [Start #208]

### PR DESCRIPTION
**Pull request by beblicarl**
#### Pre-Submission Checklist

-   [x] Your pull request targets the `main` branch.
-   [x] Branch name is descriptive and follows the guidelines [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices).
-   [x] All tests pass . Use `git commit --amend` to amend any fixes.

#### Checklist:

-   [x] All tests are passing.
-   [x] Tested changes locally.
-   [x] Improved documentation or readme where necessary.
-   [x] Your commit messages format follows our commit message guideline [here](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices#commits).

#### What does this PR do?

add table to record flagged posts
#### How should this be manually tested?

1. Start the server with `npm run dev` and you will see the migration being added to the database

#### What are the relevant issues?
add table to record flagged posts

### CODE REVIEW

**Comment by beblicarl** [comment]
@jocrah , kindly review this for me

File: migrations/1664278604452_add-table-to-record-flagged-posts.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+			references: '"users"',
+			onDelete: 'cascade'
+		},
+		postId: {
+			type: 'integer',
+			notNull: true,
+			references: '"posts"',
+			onDelete: 'cascade'
+		},
+		reason: { type: 'text', notNull: false },
```
Comment (jocrah) [inline migrations/1664278604452_add-table-to-record-flagged-posts.js:18]: `notNull` not needed if its not being set as true. This works anyway, same as without the `notNull` flag

File: migrations/1664278604452_add-table-to-record-flagged-posts.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		userId: {
+			type: 'integer',
+			notNull: true,
+			references: '"users"',
+			onDelete: 'cascade'
+		},
+		postId: {
+			type: 'integer',
+			notNull: true,
+			references: '"posts"',
```
Comment (jocrah) [inline migrations/1664278604452_add-table-to-record-flagged-posts.js:15]: now seeing this, the double quotes are not too necessary here.

**Review by jocrah — APPROVED** [review]
this works perfectly, the reviews are just to note the usage of some things that don't affect the migration

File: migrations/1664278604452_add-table-to-record-flagged-posts.js (hunk 2 above)
Comment (beblicarl) [reply migrations/1664278604452_add-table-to-record-flagged-posts.js:15]: okay, i will make the change as required

File: migrations/1664278604452_add-table-to-record-flagged-posts.js (hunk 1 above)
Comment (beblicarl) [reply migrations/1664278604452_add-table-to-record-flagged-posts.js:18]: yes please

Comment (beblicarl) [reply migrations/1664278604452_add-table-to-record-flagged-posts.js]: i have pushed the changes

File: migrations/1664278604452_add-table-to-record-flagged-posts.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+
+exports.shorthands = undefined
+
+exports.up = (pgm) => {
+	pgm.createTable('post_flags', {
+		userId: {
+			type: 'integer',
+			notNull: true,
+			references: "users",
```
Comment (jocrah) [inline migrations/1664278604452_add-table-to-record-flagged-posts.js:9]: great, let's kindly use single quotes for consistency with the other strings

**Comment by beblicarl** [comment]
I have pushed the changes @jocrah

**Pull request merged**